### ISSUE TITLE
the version of pymilvus in docker image v0.6.4 is wrong.

### ISSUE THREAD

**Issue by azuercici** [langgenius/dify#3762]
### Self Checks

- [X] This is only for bug report, if you would like to ask a quesion, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Pleas do not modify this template :) and fill in all the required fields.

### Dify version

0.6.4

### Cloud or Self Hosted

Self Hosted (Docker)

### Steps to reproduce

1. Create a knowledge base
2. Upload a file, set all policies to default
3. wait for a while, then you will see the embedding process went wrong, the message is "MilvusClient' object has no attribute 'create_collection_with_schema"

```
{"log":"Traceback (most recent call last):\n","stream":"stderr","time":"2024-04-24T01:42:01.938216941Z"}
{"log":"  File \"/app/api/core/indexing_runner.py\", line 73, in run\n","stream":"stderr","time":"2024-04-24T01:42:01.938220916Z"}
{"log":"    self._load(\n","stream":"stderr","time":"2024-04-24T01:42:01.938224432Z"}
{"log":"  File \"/app/api/core/indexing_runner.py\", line 677, in _load\n","stream":"stderr","time":"2024-04-24T01:42:01.938227176Z"}
{"log":"    tokens += future.result()\n","stream":"stderr","time":"2024-04-24T01:42:01.938230072Z"}
{"log":"  File \"/usr/local/lib/python3.10/concurrent/futures/_base.py\", line 458, in result\n","stream":"stderr","time":"2024-04-24T01:42:01.938233009Z"}
{"log":"    return self.__get_result()\n","stream":"stderr","time":"2024-04-24T01:42:01.938242685Z"}
{"log":"  File \"/usr/local/lib/python3.10/concurrent/futures/_base.py\", line 403, in __get_result\n","stream":"stderr","time":"2024-04-24T01:42:01.938245627Z"}
{"log":"    raise self._exception\n","stream":"stderr","time":"2024-04-24T01:42:01.938248568Z"}
{"log":"  File \"/usr/local/lib/python3.10/concurrent/futures/thread.py\", line 58, in run\n","stream":"stderr","time":"2024-04-24T01:42:01.938251313Z"}
{"log":"    result = self.fn(*self.args, **self.kwargs)\n","stream":"stderr","time":"2024-04-24T01:42:01.938254226Z"}
{"log":"  File \"/app/api/core/indexing_runner.py\", line 732, in _process_chunk\n","stream":"stderr","time":"2024-04-24T01:42:01.938256986Z"}
{"log":"    index_processor.load(dataset, chunk_documents, with_keywords=False)\n","stream":"stderr","time":"2024-04-24T01:42:01.9382601Z"}
{"log":"  File \"/app/api/core/rag/index_processor/processor/paragraph_index_processor.py\", line 60, in load\n","stream":"stderr","time":"2024-04-24T01:42:01.93826315Z"}
{"log":"    vector.create(documents)\n","stream":"stderr","time":"2024-04-24T01:42:01.938266198Z"}
{"log":"  File \"/app/api/core/rag/datasource/vdb/vector_factory.py\", line 147, in create\n","stream":"stderr","time":"2024-04-24T01:42:01.938269034Z"}
{"log":"    self._vector_processor.create(\n","stream":"stderr","time":"2024-04-24T01:42:01.938277105Z"}
{"log":"  File \"/app/api/core/rag/datasource/vdb/milvus/milvus_vector.py\", line 67, in create\n","stream":"stderr","time":"2024-04-24T01:42:01.938279837Z"}
{"log":"    self.create_collection(embeddings, metadatas, index_params)\n","stream":"stderr","time":"2024-04-24T01:42:01.938283204Z"}
{"log":"  File \"/app/api/core/rag/datasource/vdb/milvus/milvus_vector.py\", line 235, in create_collection\n","stream":"stderr","time":"2024-04-24T01:42:01.938286375Z"}
{"log":"    self._client.create_collection_with_schema(collection_name=collection_name,\n","stream":"stderr","time":"2024-04-24T01:42:01.938289858Z"}
{"log":"AttributeError: 'MilvusClient' object has no attribute 'create_collection_with_schema'. Did you mean: '_create_collection_with_schema'?\n","stream":"stderr","time":"2024-04-24T01:42:01.938292101Z"}

```

### ✔️ Expected Behavior

creating the knowledge base successfully

### ❌ Actual Behavior

something went wrong

**Comment by crazywoola** [comment]
@bowenliang123

**Comment by azuercici** [comment]
after searching I found that the python package pymilvus has changed the function name “create_collection_with_schema” to “_create_collection_with_schema” after v2.3.0, so the version of pymilvus in requirment.txt should be before 2.3.0（in my case I change it to 2.2.16）

**Comment by bowenliang123** [comment]
Sorry to @azuercici . It's an unexpected API change from `pymilvus` 2.3.2+ dropping the `create_collection_with_schema` method. It shall be fixed in next version, or you can rebuild the image by installing pymilvus 2.3.0 for workaround.

1. it's fixed in #3683 by replacing `create_collection_with_schema` usage to `create_collection`
2. proper unit test on vdbs are ready for merging in #3688 (not being merged yet)
